java/query/GraqlDefinable.java

```diff
@@ -76,7 +88,7 @@ public final boolean equals(final Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         final GraqlDefinable that = (GraqlDefinable) o;
-        return this.keyword.equals(that.keyword) && this.variables.equals(that.variables);
+        return this.keyword.equals(that.keyword) && this.typeVariables.equals(that.typeVariables) && this.rules.equals(that.rules);
```

Comment:
Do we not care about the order they were written now in determining their identity? Previously we did, right? If we save the initial list, and simply compare that, then we keep that behaviour.

Reply:
yeah ok can keep the original list